File: src/content/docs/topics/cs/leetcode-150/two-pointers/167-two-sum-ii.py
def two_sum(numbers: list[int], target: int) -> list[int]:
    pass  # TODO: implement

def _run_tests():
    assert two_sum([2, 7, 11, 15], 9) == [1, 2]
    assert two_sum([2, 3, 4], 6) == [1, 3]
    assert two_sum([3, 3], 6) == [1, 2]
    assert two_sum([1, 2, 3, 4, 5], 9) == [4, 5]
    assert two_sum([-3, -1, 0, 2, 4], 1) == [1, 5]
    # --- large-input timing ---
    import time as _t
    _t0 = _t.perf_counter()
    two_sum(list(range(1, 10001)), 19999)
    _ms = (_t.perf_counter() - _t0) * 1000
    print(f'perf 10000-element sorted array: {_ms:.1f}ms')
    print('all tests pass')

if __name__ == '__main__':
    _run_tests()

File: src/content/docs/topics/cs/coding-problems/two-pointers/167-two-sum-ii.swift
// LEETCODE_TYPE: Solution
func expectEqual<T: Equatable>(
    _ actual: T,
    _ expected: T,
    _ message: String = "",
    file: StaticString = #fileID,
    line: UInt = #line
) {
    guard actual == expected else {
        let detail = message.isEmpty ? "values differ" : message
        fatalError("\(file):\(line): \(detail). Expected \(expected), got \(actual)")
    }
}

func expectTrue(
    _ condition: @autoclosure () -> Bool,
    _ message: String = "expected true",
    file: StaticString = #fileID,
    line: UInt = #line
) {
    guard condition() else {
        fatalError("\(file):\(line): \(message)")
    }
}

func reportSuccess() {
    print("All Swift tests passed")
}

final class Solution {
    func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
        // TODO: Implement
        fatalError("TODO: Implement")
    }
}

func runTests() {
    // TEST_VECTORS_BEGIN sha256:3962c23c1c450f3ab66af4bac8c336a4b23468aa59897e7d04861fbc2050a1ce
    expectEqual(Solution().twoSum([2, 7, 11, 15], 9), [1, 2], "first-pair")
    expectEqual(Solution().twoSum([2, 3, 4], 6), [1, 3], "outer-pair")
    expectEqual(Solution().twoSum([-1, 0], -1), [1, 2], "negative-values")
    expectEqual(Solution().twoSum([1, 2], 3), [1, 2], "minimum-length")
    // EXCLUDED_VECTOR unsorted-input: [[3,2,4],6] | The input array must be sorted in nondecreasing order.
    // EXCLUDED_VECTOR no-solution: [[1,2,3],8] | The published problem guarantees exactly one solution.
    // TEST_VECTORS_END
    reportSuccess()
}

runTests()
